vhdbuilder/packer/install-dependencies.sh

```diff
@@ -104,179 +106,43 @@ cat << EOF >> ${VHD_LOGS_FILEPATH}
-    downloadCNI
-    echo "  - CNI plugin version ${CNI_PLUGIN_VERSION}" >> ${VHD_LOGS_FILEPATH}
-done
-
-CNI_PLUGIN_VERSIONS="
-0.8.6
-"
-for CNI_PLUGIN_VERSION in $CNI_PLUGIN_VERSIONS; do
-    CNI_PLUGINS_URL="https://acs-mirror.azureedge.net/cni-plugins/v${CNI_PLUGIN_VERSION}/binaries/cni-plugins-linux-amd64-v${CNI_PLUGIN_VERSION}.tgz"
-    downloadCNI
-    echo "  - CNI plugin version ${CNI_PLUGIN_VERSION}" >> ${VHD_LOGS_FILEPATH}
-done
-
 installImg
 echo "  - img" >> ${VHD_LOGS_FILEPATH}
 
 echo "${CONTAINER_RUNTIME} images pre-pulled:" >> ${VHD_LOGS_FILEPATH}
 
-DASHBOARD_VERSIONS="1.10.1"
-for DASHBOARD_VERSION in ${DASHBOARD_VERSIONS}; do
-    CONTAINER_IMAGE="mcr.microsoft.com/oss/kubernetes/kubernetes-dashboard:v${DASHBOARD_VERSION}"
-    pullContainerImage ${cliTool} ${CONTAINER_IMAGE}
-    echo "  - ${CONTAINER_IMAGE}" >> ${VHD_LOGS_FILEPATH}
-done
-
-NEW_DASHBOARD_VERSIONS="
-2.0.0-beta8
-2.0.0-rc3
-2.0.0-rc7
-2.0.1
-"
-for DASHBOARD_VERSION in ${NEW_DASHBOARD_VERSIONS}; do
-    CONTAINER_IMAGE="mcr.microsoft.com/oss/kubernetes/dashboard:v${DASHBOARD_VERSION}"
-    pullContainerImage ${cliTool} ${CONTAINER_IMAGE}
-    echo "  - ${CONTAINER_IMAGE}" >> ${VHD_LOGS_FILEPATH}
-done
-
-NEW_DASHBOARD_METRICS_SCRAPER_VERSIONS="
-1.0.2
-1.0.3
-1.0.4
-"
-for DASHBOARD_VERSION in ${NEW_DASHBOARD_METRICS_SCRAPER_VERSIONS}; do
-    CONTAINER_IMAGE="mcr.microsoft.com/oss/kubernetes/metrics-scraper:v${DASHBOARD_VERSION}"
-    pullContainerImage ${cliTool} ${CONTAINER_IMAGE}
-    echo "  - ${CONTAINER_IMAGE}" >> ${VHD_LOGS_FILEPATH}
-done
-
-EXECHEALTHZ_VERSIONS="1.2"
-for EXECHEALTHZ_VERSION in ${EXECHEALTHZ_VERSIONS}; do
-    CONTAINER_IMAGE="mcr.microsoft.com/oss/kubernetes/exechealthz:${EXECHEALTHZ_VERSION}"
-    pullContainerImage ${cliTool} ${CONTAINER_IMAGE}
-    echo "  - ${CONTAINER_IMAGE}" >> ${VHD_LOGS_FILEPATH}
-done
-
-ADDON_RESIZER_VERSIONS="
-1.8.5
-1.8.4
-1.8.1
-1.7
-"
-for ADDON_RESIZER_VERSION in ${ADDON_RESIZER_VERSIONS}; do
-    CONTAINER_IMAGE="mcr.microsoft.com/oss/kubernetes/autoscaler/addon-resizer:${ADDON_RESIZER_VERSION}"
-    pullContainerImage ${cliTool} ${CONTAINER_IMAGE}
-    echo "  - ${CONTAINER_IMAGE}" >> ${VHD_LOGS_FILEPATH}
-done
-
-METRICS_SERVER_VERSIONS="
-0.3.6
-0.3.5
-"
-for METRICS_SERVER_VERSION in ${METRICS_SERVER_VERSIONS}; do
-    CONTAINER_IMAGE="mcr.microsoft.com/oss/kubernetes/metrics-server:v${METRICS_SERVER_VERSION}"
-    pullContainerImage ${cliTool} ${CONTAINER_IMAGE}
-    echo "  - ${CONTAINER_IMAGE}" >> ${VHD_LOGS_FILEPATH}
-done
-
-LEGACY_MCR_PAUSE_VERSIONS="1.2.0"
-for PAUSE_VERSION in ${LEGACY_MCR_PAUSE_VERSIONS}; do
-    # Pull the arch independent MCR pause image which is built for Linux and Windows
-    CONTAINER_IMAGE="mcr.microsoft.com/k8s/core/pause:${PAUSE_VERSION}"
-    pullContainerImage ${cliTool} "${CONTAINER_IMAGE}"
-    echo "  - ${CONTAINER_IMAGE}" >> ${VHD_LOGS_FILEPATH}
-done
+string_replace() {
+  echo ${1//\*/$2}
+}
```

Comment:
what is this?

Reply:
this replaces the * in the strings with the versions